lib/collection/src/collection.rs

```diff
@@ -359,8 +363,8 @@ impl Collection {
         );
         update_handler.optimizers = new_optimizers;
         update_handler.flush_timeout_sec = config.optimizer_config.flush_interval_sec;
-        update_handler.run_workers();
-        self.update_sender.send(UpdateSignal::Nop)?;
+        update_handler.run_workers(update_receiver);
+        self.update_sender.load().send(UpdateSignal::Nop)?;
```

Comment:
So the problem was in the fact that `run_workers` was called in `update_optimizer_params` and it tried to take the old receiver from option which was empty (because it was taken on collection creation)?

Reply:
yes